lib/cartopy/tests/test_polygon.py

```diff
@@ -259,6 +259,21 @@ def test_same_points_on_boundary_2(self):
 
         assert abs(1200 - projected.area) < 1e-5
 
+    def test_attach_short_loop(self):
+        # Geometry comes from a matplotlib contourf.
+        mstring = shapely.wkt.loads(
+            'MULTILINESTRING ('
+            '(-179.9999982118607 71.87500000000001,'
+            '-179.0625 71.87500000000001,'
+            '-179.9999982118607 71.87500000000001))')
+        multi_line_strings = [mstring]
+
+        src = ccrs.PlateCarree()
+        polygons = src._attach_lines_to_boundary(multi_line_strings, True)
```

Comment:
Is there any way to do this without invoking an internal API? Leaving the test as is would hinder future attempts at refactoring and changing the non-public API.

Reply:
So here's my thinking behind writing the test as it is:
If I were to call the nearest public method (project_geometry) then on failure, the problematic shapely object will be called and this will cause a segfault. As it is written, the problematic shapely object is never called and the test fails entirely within python.
Also, this file already calls this method at line [137](https://github.com/SciTools/cartopy/pull/1317/files#diff-67bdbbb09e12e0d9d12703599512761cL137).

That said, I ought to be able to write a test (whose failure state is a segfault) which calls the public project_geometry method if that is preferable.

Reply:
Can you add the segfault test in addition? That way if we refactor, we can just ditch the old test but ensure that we're still testing the original use case and not breaking it.

Reply:
Okay, I've added the second test.